Compile/src/Producers/PUJetIDWeightProducer.cc

```diff
@@ -0,0 +1,85 @@
+#include "Excalibur/Compile/interface/Producers/PUJetIDWeightProducer.h"
+#include "KappaTools/RootTools/interface/Matching.h"
+
+std::string PUJetIDWeightProducer::GetProducerId() const { return "PUJetIDWeightProducer"; }
+
+void PUJetIDWeightProducer::Init(ZJetSettings const& settings) {
+
+    PUJetID = settings.GetPUJetID();
+    PUJetIDSelectionMaxPt = 50.0;
+    inputIsData = settings.GetInputIsData();
+
+    if (inputIsData)
+    {
+        LOG(FATAL) << "PUJetID SFs can only be applied to MC!";
+    }
+
+    if (PUJetID != "none")
+    {
+        LOG(INFO) << "Apply PUJetID cut on jets pt<" << PUJetIDSelectionMaxPt << " GeV";
+        PUEffFilename = settings.GetPUJetIDEffFilename();
+        std::string histName = settings.GetPUJetIDEffHistogramName();
+        TFile PUEffFile(PUEffFilename.c_str(), "READONLY");
+        if (PUEffFile.IsOpen() == false)
+        {
+            LOG(FATAL) << "File for PUJetID scalefactors (" << PUEffFilename <<
+                " not found. Please check PUEffFilename.";
+        }
+
+        LOG(INFO) << "load histogram " << histName << " for PUJetID scalefactors";
+        sfhisto = (TH2F*)PUEffFile.Get(histName.c_str());
+        if (!sfhisto)
+        {
+            LOG(FATAL)<<"no histogram found with name" << histName;
+        }
+        sfhisto->SetDirectory(0);
+
+        std::string errName = histName+"_Systuncty";
```

Comment:
Are you sure this naming scheme is persistent? Maybe you just want to add another option in the settings.

Reply:
Fixed